File: API/API/Models/Project.vb
Imports System.Collections.Generic

Public Class Project
    Public Property Id As Guid
    Public Property Name As String
    Public Property Description As String
    Public Property OwnerId As Guid
    Public Property CreatedAt As DateTime
    Public Property UpdatedAt As DateTime

    Public Overridable Property Owner As User
    Public Overridable Property Tasks As ICollection(Of TodoTask)
    Public Overridable Property ProjectMembers As ICollection(Of ProjectMember)
End Class

File: API_v2/Models/Project.cs
using System;
using System.Collections.Generic;

namespace API_v2.Models
{
    public class Project
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public Guid OwnerId { get; set; }
        public string? GoogleDriveFolderId { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        // Navigation properties
        public virtual User Owner { get; set; } = null!;
        public virtual ICollection<TodoTask> Tasks { get; set; } = new List<TodoTask>();
        public virtual ICollection<ProjectMember> ProjectMembers { get; set; } = new List<ProjectMember>();
        public virtual ICollection<ProjectColumn> Columns { get; set; } = new List<ProjectColumn>();
        public virtual ICollection<ProjectFile> Files { get; set; } = new List<ProjectFile>();
        public virtual ICollection<ProjectFolder> Folders { get; set; } = new List<ProjectFolder>();
        public virtual ICollection<ProjectFileActivity> FileActivities { get; set; } = new List<ProjectFileActivity>();
    }
}
